src/vim9class.c

```diff
@@ -610,22 +610,24 @@ check_func_arg_names(
     static int
 is_duplicate_member(garray_T *mgap, char_u *varname, char_u *varname_end)
 {
-    char_u *pstr = (*varname == '_') ? varname + 1 : varname;
+    char_u	*name = vim_strnsave(varname, varname_end - varname);
+    char_u	*pstr = (*name == '_') ? name + 1 : name;
+    int		dup = FALSE;
 
     for (int i = 0; i < mgap->ga_len; ++i)
     {
 	ocmember_T *m = ((ocmember_T *)mgap->ga_data) + i;
 	char_u *qstr = *m->ocm_name == '_' ? m->ocm_name + 1 : m->ocm_name;
-	if (STRNCMP(pstr, qstr, varname_end - pstr) == 0)
```

Comment:
The pattern I usally see is something like
```
if (STRNCMP(pstr, qstr, varname_end - pstr) == 0
        && *(pstr + varname_end - pstr) == NULL)
```
much cheaper.

Reply:
I didn't remember the pattern quite correctly, a quick look in vim9compile showed several ways to check if the match is good, depending on what is known about the args. The check I showed is definitely not correct.